### PR TITLE
feat: Adding plugin for maxlength countdown for text based inputs

### PR DESCRIPTION
**Pull request by chrisladams**

### CODE REVIEW

File: packages/addons/src/plugins/maxlengthCountdown.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+          node.on('input', updateCountValue)
+
+          function updateCountValue ({ payload }: { payload: string }) {
+            let stringLength = parseInt(node.props.attrs.maxlength) - (payload ? payload.length : 0)
+            node.props.maxlengthStringCount = Math.max(0, stringLength)
+            node.props.maxlengthRemainingText = node.props.maxlengthRemainingText ||
+              MaxlengthCountdownOptions?.remainingText ||
+              'remaining characters'
+          }
+        }
```
Comment (justin-schroeder) [inline packages/addons/src/plugins/maxlengthCountdown.ts:115]: The same memoization technique discussed in the review of #1000 could apply here too. Let’s do as little work inside this input event as possible.

File: packages/addons/src/plugins/maxlengthCountdown.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      })
+    }
+  }
+
+  return maxlengthCountdownPlugin
+}
```
Comment (justin-schroeder) [inline packages/addons/src/plugins/maxlengthCountdown.ts:121]:
```suggestion
}

```

File: packages/addons/src/plugins/maxlengthCountdown.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+ * @public
+ */
+export function createMaxLengthCountdownPlugin(
+  MaxlengthCountdownOptions?: MaxlengthCountdownOptions
+): FormKitPlugin {
+
+  const maxlengthCountdownPlugin = (node: FormKitNode) => {
+    node.addProps(['maxlengthCountdown'])
+    node.addProps(['maxlengthRemainingText'])
+    node.addProps(['maxlengthStringCount'])
```
Comment (justin-schroeder) [inline packages/addons/src/plugins/maxlengthCountdown.ts:42]:
```suggestion
    node.addProps(['maxlengthCountdown', 'maxlengthRemainingText', 'maxlengthStringCount'])
```

**Review by justin-schroeder — CHANGES_REQUESTED** [review]
@chrisladams small little changes, looks great overall.

File: packages/addons/src/plugins/maxlengthCountdown.ts (hunk 3 above)
Comment (chrisladams) [reply packages/addons/src/plugins/maxlengthCountdown.ts]: Updated

**Comment by chrisladams** [comment]
@justin-schroeder updated! Thanks for the review

**Pull request merged**